### PR TITLE
bump dugite to 2.18.0 and related packaging improvements

### PR DESCRIPTION
**Pull request by shiftkey**
This is a rollup of a number of changes to our Git library in the last few weeks:

 - Git 2.18.0
 - Git LFS 2.5.0
 - improved error parsing for "not a repository" code path - #5302
 - embedded Git on Linux now built against OpenSSL - https://github.com/shiftkey/desktop/issues/21
 - support for installing and using on 32-bit Windows #1214

Fixes: #5302

TODO:

 - [x] investigate failing test about Git no longer reporting files as `copied`

### CODE REVIEW

**Comment by shiftkey** [comment]
For reference, Git 2.18.0 put copy detection in `git status` behind a config flag - see https://github.com/git/git/commit/e8b2dc2c2a8415bfd180ecf5cc237a54e69ac2e9 for details about the change.

File: app/src/lib/git/for-each-ref.ts [hunk 1]
```diff
@@ -45,10 +47,10 @@ export async function getBranches(
     ['for-each-ref', `--format=${format}`, ...prefixes],
     repository.path,
     'getBranches',
-    { successExitCodes: new Set([0, 128]) }
+    { expectedErrors: new Set([GitError.NotAGitRepository]) }
```
Comment (outofambit) [inline app/src/lib/git/for-each-ref.ts]: 🎉

File: app/test/unit/git/status-test.ts [hunk 2]
```diff
@@ -70,6 +70,14 @@ describe('git/status', () => {
       const testRepoPath = await setupFixtureRepository('copy-detection-status')
       repository = new Repository(testRepoPath, -1, null, false)
 
+      // Git 2.18 now uses a new config value to handle detecting copies, so
```
Comment (outofambit) [inline app/test/unit/git/status-test.ts]: ❇️

**Comment by outofambit** [comment]
holding off on merging this for 1.4.0

**Pull request merged**